packages/hash/frontend/src/pages/login.page.tsx

```diff
@@ -1,39 +1,212 @@
 import { useRouter } from "next/router";
-import { LoginModal } from "./login.page/login-modal";
-import { useUser } from "../components/hooks/useUser";
-import {
-  isParsedInvitationEmailQuery,
-  isParsedInvitationLinkQuery,
-} from "./shared/auth-utils";
+import React, { useEffect, FormEventHandler, useState, useMemo } from "react";
+import { SelfServiceLoginFlow } from "@ory/client";
+import { Typography, Container, Box, TextField } from "@mui/material";
+import { isUiNodeInputAttributes } from "@ory/integrations/ui";
+import { AxiosError } from "axios";
 import { getPlainLayout, NextPageWithLayout } from "../shared/layout";
+import {
+  createFlowErrorHandler,
+  mustGetCsrfTokenFromFlow,
+  oryKratosClient,
+  useLogoutFlow,
+} from "./shared/ory-kratos";
+import { Button } from "../shared/ui";
 
-const Page: NextPageWithLayout = () => {
-  const { refetch } = useUser();
+const LoginPage: NextPageWithLayout = () => {
+  // Get ?flow=... from the URL
   const router = useRouter();
 
-  return (
-    <LoginModal
-      show
-      onLoggedIn={({ accountSignupComplete, accountId }) => {
-        void refetch().then(() => {
-          // redirect to invite page if login occured from invitation link
-          if (accountSignupComplete) {
-            if (
-              isParsedInvitationEmailQuery(router.query) ||
-              isParsedInvitationLinkQuery(router.query)
-            ) {
-              void router.push({ pathname: "/invite", query: router.query });
+  const {
+    return_to: returnTo,
+    flow: flowId,
+    // Refresh means we want to refresh the session. This is needed, for example, when we want to update the password
+    // of a user.
+    refresh,
+    // AAL = Authorization Assurance Level. This implies that we want to upgrade the AAL, meaning that we want
+    // to perform two-factor authentication/verification.
+    aal,
+  } = router.query;
+
+  const [flow, setFlow] = useState<SelfServiceLoginFlow>();
+
+  const [email, setEmail] = useState<string>("");
```

Comment:
Are we okay with using empty strings here? A little worried we could accidentally end up with empty emails

Reply:
This is the state for the form value for email, so it needs to be initialised to something when the input is empty (i.e. `""`).

It's not possible for the empty string to be assigned as an email becuase:
1. We're using HTML form validation that prevents a required field from being submitted (and ensures that this is a valid email because the input has `type="email"`)
2. In the `handleSubmit` method we are ensuring both the email and password aren't empty on line 70 in this file
3. The kratos API does additional validation server-side (ensuring that an email + password was provided, that they are formatted correctly, and even rejects some bad passwords such as `"password"`, `"12345"`, etc). You can find out more about this [here](https://www.ory.sh/docs/kratos/self-service/flows/user-registration#registration-with-usernameemail-and-password-1)

Reply:
I've made the validation stricter so that `blah@blah` wouldn't qualify as an email anymore. I don't think there's anything left to address in this comment thread.